opnode/test/system_test.go

```diff
@@ -413,3 +417,134 @@ func TestMissingBatchE2E(t *testing.T) {
 	require.Nil(t, err, "Get block from sequencer")
 	require.NotEqual(t, block.Hash(), receipt.BlockHash, "L2 Sequencer did not reorg out transaction on it's safe chain")
 }
+
+func L1InfoFromState(ctx context.Context, contract *l1block.L1Block, l2Number *big.Int) (derive.L1BlockInfo, error) {
+	var err error
+	var out derive.L1BlockInfo
+	opts := bind.CallOpts{
+		BlockNumber: l2Number,
+		Context:     ctx,
+	}
+
+	number, err := contract.Number(&opts)
+	if err != nil {
+		return derive.L1BlockInfo{}, fmt.Errorf("failed to get number: %w", err)
+	}
+	if !number.IsUint64() {
+		return derive.L1BlockInfo{}, errors.New("number does not fit in a uint64")
+
+	}
+	out.Number = number.Uint64()
+
+	time, err := contract.Timestamp(&opts)
+	if err != nil {
+		return derive.L1BlockInfo{}, fmt.Errorf("failed to get timestamp: %w", err)
+	}
+	if !time.IsUint64() {
+		return derive.L1BlockInfo{}, errors.New("time does not fit in a uint64")
+
+	}
+	out.Time = time.Uint64()
+
+	out.BaseFee, err = contract.Basefee(&opts)
+	if err != nil {
+		return derive.L1BlockInfo{}, fmt.Errorf("failed to get timestamp: %w", err)
+	}
+
+	blockHashBytes, err := contract.Hash(&opts)
+	if err != nil {
+		return derive.L1BlockInfo{}, fmt.Errorf("failed to get block hash: %w", err)
+	}
+	out.BlockHash = common.BytesToHash(blockHashBytes[:])
+
+	return out, nil
+}
+
+func TestL1InfoContract(t *testing.T) {
+	log.Root().SetHandler(log.DiscardHandler()) // Comment this out to see geth l1/l2 logs
```

Comment:
Perhaps we can put this behind a flag that turns logs on and off and use the `-args` flag of `go test`?

https://pkg.go.dev/cmd/go/internal/test

Reply:
Out of scope of this PR, just a thought that could apply across the test suite